frontend/src/metabase/entities/data-apps/utils.ts

```diff
@@ -16,3 +22,45 @@ export function getDataAppHomePageId(dataApp: DataApp, pages: Dashboard[]) {
   const [firstPage] = _.sortBy(pages, "name");
   return firstPage?.id;
 }
+
+function isParentPage(
+  targetPageIndent: number,
+  maybeParentNavItem: DataAppNavItem,
+) {
+  if (maybeParentNavItem.hidden) {
+    return false;
+  }
+
+  // For `indent: 1` the expected parent indent is 0.
+  // But it can be coming as `undefined` from the API,
+  // so we need to ensure we accept both `undefined` and 0.
+  if (targetPageIndent === 1) {
+    return !maybeParentNavItem.indent || maybeParentNavItem.indent === 0;
+  }
+
+  return maybeParentNavItem.indent === targetPageIndent - 1;
+}
+
+export function getParentDataAppPageId(
```

Comment:
This data structure that using `indent` that requires us to calculate parent page ids has me a little concerned here. I wonder if it could get quite complex + computationally expensive for moving and nesting operations?

Reply:
Good call. I don't think we'd get into any performance trouble because of that (not sure we'd have to deal with moving too many pages too often). Let's keep an eye on that though